src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -1255,6 +1255,7 @@ private void EmitResource(ExpressionEmitter emitter, ImmutableArray<ExtensionExp
                 }
 
                 if (metadata.IsAzResource ||
+                    Context.SemanticModel.TargetScope == ResourceScope.DesiredStateConfiguration ||
```

Comment:
I think it would be a good idea to start a test file similar to https://github.com/Azure/bicep/blob/main/src/Bicep.Core.IntegrationTests/Extensibility/RadiusCompatibilityTests.cs (which was initially created to make sure we maintain compatibility with [Radius](https://radapp.io/) - to make sure that we don't regress this in future.

Happy to merge this change without it as it's ultimately an experimental feature, but just be aware that you do run the risk of getting accidentally broken again if you don't have a way to assert compatibility.